src/Backend/Core/Engine/Header.php

```diff
@@ -183,7 +189,9 @@ public function addJS(
         bool $addTimestamp = false,
         Priority $priority = null
     ): void {
-        $module = $module ?? $this->url->getModule();
+        $isExternalUrl = $this->get('fork.validator.url')->isExternalUrl($file);
+        $overwritePath = $overwritePath || $isExternalUrl;
```

Comment:
I would add a comment that explains that if the given path is an external url it always overwrites the path.